### PR TITLE
test: Improve `duration_attribute_tests`

### PR DESCRIPTION
**Pull request by luke396**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [x] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.

- Add constructor_lazy and ensure it fails as expected.
- Combine two separate tests into one.

### CODE REVIEW

File: tests/expr_and_series/dt/datetime_duration_test.py [hunk 1]
```diff
@@ -23,79 +23,54 @@
         timedelta(milliseconds=1, microseconds=300),
     ],
     "c": np.array([None, 20], dtype="timedelta64[ns]"),
-    "d": [
```
Comment (luke396) [inline tests/expr_and_series/dt/datetime_duration_test.py:26]: This appears to be unused, so it is removed.

**Comment by luke396** [comment]
The combination of two tests is somewhat aggressive; if it's not suitable, I would like to revert them.

**Review by FBruzzesi — COMMENTED** [review]
Thanks for your PR, in particular for cleaning up dead code!

The idea of moving to `constructor` instead of `constructor_eager` is to be able to assess lazy  instances for expressions, but not for series.

If you check other tests that do this, they split into:

- one test for `constructor` where expression are tested
- one test for `constructor_eager` where series are tested

[Example `.abs()`](https://github.com/narwhals-dev/narwhals/blob/main/tests/expr_and_series/abs_test.py).

I would suggest to follow a similar approach to make sure that polars lazyframe (and dask lazyframe in the future) can perform those operations.

**Review by MarcoGorelli — APPROVED** [review]
thanks @luke396 !

**Pull request merged**